src/api/verification_case.py

```diff
@@ -0,0 +1,458 @@
+                            ):
+                                if verbose:
+                                    logging.error(
+                                        f"The type of '{idf_key}' variable's value must be str, but {type(instance[key][idf_key].get('variable'))} is provided."
+                                    )
+                                return False
+
+                            if not instance[key][idf_key].get("frequency"):
+                                if verbose:
+                                    logging.error(
+                                        f"'frequency' key doesn't exist in {instance[key][idf_key]}. Please make sure to include the 'frequency' key."
+                                    )
+                                return False
+
+                            if not isinstance(
+                                instance[key][idf_key].get("frequency"), str
+                            ):
+                                if verbose:
+                                    logging.error(
+                                        f"The type of '{idf_key}' frequency's value must be str, but {type(instance[key][idf_key].get('frequency'))} is provided."
+                                    )
+                                return False
+
+                    if key == "parameters":
+                        for idf_key, idf_value in instance[key].items():
+                            if not isinstance(
+                                instance[key][idf_key], float
+                            ) and not isinstance(instance[key][idf_key], int):
+                                if verbose:
+                                    logging.error(
+                                        f"The type of '{idf_key}' value must be either float or int, but {type(instance[key][idf_key])} is provided."
+                                    )
+                                return False
+
+            return is_cases_valid
+
+        # check case type
+        if not isinstance(case, Dict):
+            logging.error(
+                f"The case argument type must be dict, but {type(case)} is provided."
+            )
+            return None
+
+        # check verbose type
+        if not isinstance(verbose, bool):
+            logging.error(
+                f"The verbose argument type must be bool, but {type(verbose)} is provided."
+            )
+            return None
+
+        # case schema used for
+        case_schema = {
+            "no": int,
+            "run_simulation": bool,
+            "simulation_IO": {
+                "idf": str,
+                "idd": str,
+                "weather": str,
+                "output": str,
+                "ep_path": str,
+            },
+            "expected_result": str,
+            "datapoints_source": {"idf_output_variables": dict, "parameters": dict},
+            "verification_class": str,
+        }
```

Comment:
The dict is used in the ```_validate_case_structure_helper``` method to help examine each value's type.

Reply:
I understand that. Two points for you to consider. 1) If it is only used in _validate_case_structure_helper, compared with defining it here and serving it to _validate_case_structure_helper as a parameter, maybe moving it inside _validate_case_structure_helper (thus no need to pass in a parameter) will be better ; 2) we likely would want to change schema structure requirements, defining it in a separate json file will make updating it easier than modifying python code.

Reply:
I agree with both (1) and (2). Let me put this JSON in the ```schema``` directory.

Reply:
Updates

Originally, I was going to create a JSON file for this, but I decided not to.
Reason)
1. in JSON file, values should be str (```"int"``` instead of ```int```). For this reason, there has to be one more for-loop for type conversion.
2. In the future, I believe this part will be covered with JSON case schema that we will develop in the near future so, we will eliminate this part anyway later.

Reply:
sounds great!